packages/flutter_test/lib/src/controller.dart

```diff
@@ -265,6 +335,304 @@ class SemanticsController {
 
     return false;
   }
+
+  /// Performs the given [SemanticsAction] on the [SemanticsNode] found by `finder`.
+  ///
+  /// If `args` are provided, they will be passed unmodified with the `action`.
+  /// The `checkForAction` argument allows for attempting to perform `action` on
+  /// `node` even if it doesn't report supporting that action. This is useful
+  /// for implicitly supported actions such as [SemanticsAction.showOnScreen].
+  void performAction(FinderBase<SemanticsNode> finder, SemanticsAction action, {Object? args, bool checkForAction = true}) {
+    final SemanticsNode node = finder.evaluate().single;
+    if (checkForAction && !node.getSemanticsData().hasAction(action)){
+      throw StateError(
+        'The given node does not support $action.\n'
+        'Node: $node'
+      );
+    }
+
+    node.owner!.performAction(node.id, action, args);
+  }
+
+  /// Performs a [SemanticsAction.tap] action on the [SemanticsNode] found
+  /// by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.tap]
+  void tap(FinderBase<SemanticsNode> finder) {
+    performAction(finder, SemanticsAction.tap);
+  }
+
+  /// Performs a [SemanticsAction.longPress] action on the [SemanticsNode] found
+  /// by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.longPress]
+  void longPress(FinderBase<SemanticsNode> finder) {
+    performAction(finder, SemanticsAction.longPress);
+  }
+
+  /// Performs a [SemanticsAction.scrollLeft] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.scrollLeft]
+  void scrollLeft({FinderBase<SemanticsNode>? finder}) {
```

Comment:
Is this a _named_ optional parameter on purpose? Or should it be an unnamed one? It feels a little odd, that you can call `longPress(myFinder)`, but have to do `scrollLeft(finder: myFinder)` instead of `scrollLeft(myFinder)`.